packages/components/addon/modifiers/hds-tooltip.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Modifier from 'ember-modifier';
+import { assert } from '@ember/debug';
+import { registerDestructor } from '@ember/destroyable';
+
+import tippy, { followCursor } from 'tippy.js';
+
+function cleanup(instance) {
+  const { interval, needsTabIndex, tooltip } = instance;
+  if (needsTabIndex) {
+    tooltip?.reference?.removeAttribute('tabindex');
+  }
+  clearInterval(interval);
+  tooltip?.destroy();
+}
+
+/**
+ *
+ * `Tooltip` implements a modifier that uses Tippy.js to display a tooltip.
+ *
+ * Sample usage:
+ * ```
+ * <div {{hds-tooltip 'Text' options=(hash )}}>Hover me!</div>
+ * ```
+ *
+ * @see https://atomiks.github.io/tippyjs
+ * @class TooltipModifier
+ *
+ */
+export default class HdsTooltipModifier extends Modifier {
+  didSetup = false;
+
+  interval = null;
+  needsTabIndex = false;
+  tooltip = null;
+
+  constructor(owner, args) {
+    super(owner, args);
+    registerDestructor(this, cleanup);
+  }
+
+  modify(element, positional, named) {
+    assert('Tooltip must have an element', element);
+
+    if (!this.didSetup) {
+      this.#setup(element, positional, named);
+      this.didSetup = true;
+    }
+
+    this.#update(element, positional, named);
+  }
+
+  #setup(element, positional, named) {
```

Comment:
are we using the `#` used to denote a private class? we've not done this so far – or if we did it we may be using `_` (I'll have to check the codebase)

Reply:
I'm not sure because I haven't looked closely at this code yet. It's just taken from Cloud-ui. At this stage I'm looking for feedback on general approaches vs. syntax, etc. Please see the comments I made on the code I pushed up asking for feedback.

Reply:
I'm happy to modify this code however matches our general approach btw, I just haven't done much with it so far and looking for higher level feedback to begin.

Reply:
I found this reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

"Class fields are [public](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields) by default, but private class members can be created by using a hash # prefix. The privacy encapsulation of these class features is enforced by JavaScript itself."

Reply:
ok, so they're standard private class fields, no Ember 'coating' – that sounds ok to me; we may want to extend the use of this to other private fields across our components, something for the engineering sync